File: php/src/pattern/strategy/duck/RubberDuck.php
<?php

namespace pattern\strategy\duck;

class RubberDuck extends Duck
{
    public function __construct()
    {
	parent::__construct();
	$this->_flyBehaviour = new FlyNoWay();
	$this->_quackBehaviour = new Squeak();
    }

    public function display()
    {
	echo "I'm a cute yellow rubber duck...\n";
    }
}

File: java/src/pattern/strategy/duck/RubberDuck.java
package pattern.strategy.duck;

public class RubberDuck extends Duck
{
    public RubberDuck()
    {
	super.Duck();
	flyBehaviour = new FlyNoWay();
	quackBehaviour = new Squeak();
    }

    public void display()
    {
	System.out.println("I'm a cute yellow rubber duck...");
    }
}
